src/CellularPanel.cpp

```diff
@@ -962,29 +961,30 @@ void CellularPanel::HandleClick( const TrackPanelMouseEvent &tpmEvent )
          HandleMotion( tpmState );
       }
       ProcessUIHandleResult(
-         pCell.get(), pCell.get(), refreshResult);
+         pCell, pCell, refreshResult);
       state.mMouseOverUpdateFlags |= refreshResult;
    }
 }
 
-void CellularPanel::DoContextMenu( TrackPanelCell *pCell )
+void CellularPanel::DoContextMenu( const std::shared_ptr<TrackPanelCell> &pCell )
```

Comment:
I would simplify this function:  let its `pCell` argument be a non-reference -- which implies a copying of a shared pointer when it is called.  So the explicit copying when you make `pCellLocal` is instead implicit.  And then you can simply reassign `pCell` as needed.  Fewer lines of code would change.

Reply:
Thanks Paul.  I like this suggestion.  I'll make the change, retest, and commit the changes into the pull request.  Stay tuned.